client/src/components/Sidebar/ChatContent.js

```diff
@@ -18,24 +18,63 @@ const useStyles = makeStyles((theme) => ({
     color: "#9CADC8",
     letterSpacing: -0.17,
   },
+  unreadMessageBox: {
+    display: "flex",
+    justifyContent: "center",
+    alignItems: "center",
+    height: "100%",
+    margin: "auto 0",
+  },
+  unreadBadge: {
+    height: "20px",
+    width: "20px",
+    borderRadius: "50%",
+    top: "10px",
+    right: "20px",
+  },
 }));
 
+const countUnReadMessage = (messages, otherUserId) =>
+  messages.reduce(
+    (acc, message) =>
+      acc +
+      (message.readStatus === false && message.senderId === otherUserId
+        ? 1
+        : 0),
+    0
+  );
+
 const ChatContent = (props) => {
   const classes = useStyles();
 
-  const { conversation } = props;
+  const { conversation, setUnReadMessage, unReadMessage } = props;
   const { latestMessageText, otherUser } = conversation;
 
+  useEffect(() => {
+    if (!setUnReadMessage || !conversation) return;
+    setUnReadMessage(
+      countUnReadMessage(conversation.messages, conversation.otherUser.id)
+    );
+  }, [conversation, setUnReadMessage]);
```

Comment:
The problem with this is that everytime conversation changes (i.e. when a new message gets send/received) it will recalculate all the unread messages, whereas if the unreadCount was saved in the store state, and only calculated once at the time of fetching, then you would only have to increment that number every time a new message comes in and you'll save a lot of unnecessary computation 🙂